File: cplusplus/Product.cpp
#include "Product.hpp"

Product::Product(std::string name, bool isNeedPrepare) : name(name), isNeedPrepare(isNeedPrepare)
{
}

std::string Product::getName()
{
	return name;
}
bool Product::getIsNeedPrepare()
{
	return isNeedPrepare;
}

std::string Product::toString()
{
	return "Product(" + getHashCode() + ")" + " name=" + name;
}

std::string Product::getHashCode()
{
	std::string thisAddress = std::to_string(reinterpret_cast<int>(this));
	std::string hashCode = std::to_string(std::hash<std::string>()(thisAddress));

	return hashCode;
}

File: python/root/Product.py
class Product:
    __name: str
    __isNeedPrepare: bool

    def __init__(self, name: str, isNeedPrepare: bool):
        self.__name = name
        self.__isNeedPrepare = isNeedPrepare

    def getName(self) -> str:
        return self.__name

    def isNeedPrepare(self) -> bool:
        return self.__isNeedPrepare

    def __repr__(self) -> str:
        return f'Product({self.__hash__()}), name: {self.__name}'
